native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -1909,14 +1916,29 @@ static proxy_worker *internal_find_best_byrequests(proxy_balancer *balancer, pro
              * not in error state or not disabled.
              * and that can map the context.
              */
-            if (node_storage->read_node(worker->s->index, &node) != APR_SUCCESS)
-                continue; /* Can't read node */
-            pptr = (char *) node;
-            pptr = pptr + node->offset;
-            if (worker->s != (proxy_worker_shared *) pptr)
-                continue; /* wrong shared memory address */
+            if (PROXY_WORKER_IS_USABLE(worker)) {
+                if (best == NULL)
+                    return NULL;
```

Comment:
that will break the logic (well the trace and a set), better break here.

Reply:
This is moving context_host_ok logic into internal_find_best_byrequests.  context_host_ok already returned null if best from find_node_context_host is null here.  So instead of repeating context_host_ok/find_node_context_host each iteration, find_node_context_host is called just once.  So if best is null in one iteration, it would be null in all if we did a break to repeat iterations.  Thus it seems we should just return here.

Reply:
We missed the piece https://github.com/modcluster/mod_cluster/blob/e897423c6df4089cb23f2584512dfb9e0a6465b1/native/mod_proxy_cluster/mod_proxy_cluster.c#L2034 no?

Reply:
Ah, yes; sorry, see what you mean now.